--- ./Makefile.orig	2021-02-20 12:47:42.000000000 +0800
+++ ./Makefile	2021-02-20 12:48:03.000000000 +0800
@@ -72,12 +72,10 @@
 
 check_go_env:
 	@(hash go) || (echo "\nERROR: GO environment not setup properly!\n"; exit 1)
-	@go version | grep -q $(GO_VERSION) || (echo "\nERROR: Please upgrade your GO version\nThis project requires version $(GO_VERSION)"; exit 1)
 .PHONY: check_go_env
 
 check_node_env:
 	@(hash node) || (echo "\nERROR: Node environment not setup properly!\n"; exit 1)
-	@node --version | grep -q $(NODE_VERSION) || (echo "\nERROR: Please check your Node version. Should be $(NODE_VERSION)\n"; exit 1)
 .PHONY: check_node_env
 
 build: check_go_env
