diff --git a/Makefile b/Makefile
index 59e1da6d4..3c810b294 100644
--- a/Makefile
+++ b/Makefile
@@ -10,6 +10,7 @@ GIT_TAG=$(patsubst navidrome-%,v%,$(notdir $(PWD)))
 endif
 
 CI_RELEASER_VERSION=1.17.2-1 ## https://github.com/navidrome/ci-goreleaser
+GOPROXY=https://proxy.golang.org/
 
 setup: check_env download-deps setup-git ##@1_Run_First Install dependencies and prepare development environment
 	@echo Downloading Node dependencies...
@@ -111,7 +112,7 @@ release:
 
 download-deps:
 	@echo Downloading Go dependencies...
-	@go mod download -x
+	@GOPROXY=$(GOPROXY) go mod download -x
 	@go mod tidy # To revert any changes made by the `go mod download` command
 .PHONY: download-deps
 

From c0364331748b8e0600b27caa956caa8160b2bdcd Mon Sep 17 00:00:00 2001
From: Pim Zandbergen <pim@zandbergen.org>
Date: Sun, 12 Jun 2022 21:48:34 +0200
Subject: [PATCH 2/2] add support for OpenSSL 3.0

---
 Makefile                            |  1 +
 ui/webpack-patch-for-openssl3.patch | 11 +++++++++++
 2 files changed, 12 insertions(+)
 create mode 100644 ui/webpack-patch-for-openssl3.patch

diff --git a/Makefile b/Makefile
index 3c810b294..d7ac83908 100644
--- a/Makefile
+++ b/Makefile
@@ -15,6 +15,7 @@ GOPROXY=https://proxy.golang.org/
 setup: check_env download-deps setup-git ##@1_Run_First Install dependencies and prepare development environment
 	@echo Downloading Node dependencies...
 	@(cd ./ui && npm ci)
+	@(cd ./ui && grep -q crypto node_modules/react-scripts/config/webpack.config.js || patch -p0 < webpack-patch-for-openssl3.patch)
 .PHONY: setup
 
 dev: check_env   ##@Development Start Navidrome in development mode, with hot-reload for both frontend and backend
diff --git a/ui/webpack-patch-for-openssl3.patch b/ui/webpack-patch-for-openssl3.patch
new file mode 100644
index 000000000..fa1f3fe63
--- /dev/null
+++ b/ui/webpack-patch-for-openssl3.patch
@@ -0,0 +1,11 @@
+--- node_modules/react-scripts/config/webpack.config.js.orig	2022-06-12 21:14:45.411754951 +0200
++++ node_modules/react-scripts/config/webpack.config.js	2022-06-12 21:14:19.397151014 +0200
+@@ -796,3 +796,8 @@
+     performance: false,
+   };
+ };
++
++// fix for OpenSSL 3.0, that no longer supports MD4
++const crypto = require("crypto");
++const crypto_orig_createHash = crypto.createHash;
++crypto.createHash = algorithm => crypto_orig_createHash(algorithm == "md4" ? "sha256" : algorithm);
