diff --git a/Makefile b/Makefile
index 28d34d42..2ce33457 100644
--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ CI_RELEASER_VERSION ?= 1.23.0-1 ## https://github.com/navidrome/ci-goreleaser
 
 UI_SRC_FILES := $(shell find ui -type f -not -path "ui/build/*" -not -path "ui/node_modules/*")
 
-setup: check_env download-deps setup-git ##@1_Run_First Install dependencies and prepare development environment
+setup: ##@1_Run_First Install dependencies and prepare development environment
 	@echo Downloading Node dependencies...
 	@(cd ./ui && npm ci)
 .PHONY: setup
@@ -80,8 +80,8 @@ setup-git: ##@Development Setup Git hooks (pre-commit and pre-push)
 	@(cd .git/hooks && ln -sf ../../git/* .)
 .PHONY: setup-git
 
-build: check_go_env buildjs ##@Build Build the project
-	go build -ldflags="-X github.com/navidrome/navidrome/consts.gitSha=$(GIT_SHA) -X github.com/navidrome/navidrome/consts.gitTag=$(GIT_TAG)-SNAPSHOT" -tags=netgo
+build: buildjs ##@Build Build the project
+	go build -ldflags="-X github.com/navidrome/navidrome/consts.gitSha=$(GIT_SHA) -X github.com/navidrome/navidrome/consts.gitTag=$(GIT_TAG)" -tags=netgo
 .PHONY: build
 
 buildall: deprecated build
@@ -91,7 +91,7 @@ debug-build: check_go_env buildjs ##@Build Build the project (with remote debug
 	go build -gcflags="all=-N -l" -ldflags="-X github.com/navidrome/navidrome/consts.gitSha=$(GIT_SHA) -X github.com/navidrome/navidrome/consts.gitTag=$(GIT_TAG)-SNAPSHOT" -tags=netgo
 .PHONY: debug-build
 
-buildjs: check_node_env ui/build/index.html ##@Build Build only frontend
+buildjs: ui/build/index.html ##@Build Build only frontend
 .PHONY: buildjs
 
 ui/build/index.html: $(UI_SRC_FILES)
diff --git a/ui/src/dialogs/AboutDialog.js b/ui/src/dialogs/AboutDialog.js
index 046e2eb7..ac0f44bf 100644
--- a/ui/src/dialogs/AboutDialog.js
+++ b/ui/src/dialogs/AboutDialog.js
@@ -38,7 +38,7 @@ const LinkToVersion = ({ version }) => {
     ? `https://github.com/navidrome/navidrome/compare/v${
         parts[0].split('-')[0]
       }...${commitID}`
-    : `https://github.com/navidrome/navidrome/releases/tag/v${parts[0]}`
+    : `https://github.com/navidrome/navidrome/releases/tag/v${parts[0].split('-')[0]}`
   return (
     <TableCell align="left">
       <Link href={url} target="_blank" rel="noopener noreferrer">
